### PR TITLE
New JSON Syntax cheat sheet

### PR DESCRIPTION
**Pull request by vCillusion**
IA: https://duck.co/ia/view/json_syntax_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@vCillusion Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	],
+	"sections": {
+		"JSON Elements": [{
+			"val": "string / number / object / array / true / false / null / undefined",
+			"key": "value"
+		}, {
+			"val": "value / value , elements",
+			"key": "elements"
+		}, {
+			"val": "[] / [ elements ]",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: you need to change  ] and [  to `\\]` and `\\[` respectively, Please update it,

Just change this to `"val": "\\[\\] / [ elements ]",`

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			"key": "array"
+		}, {
+			"val": "string : value",
+			"key": "pair"
+		}, {
+			"val": "pair / pair , members",
+			"key": "members"
+		}, {
+			"val": "{} / { members }",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: this also needs ti be escaped:
`"val": "\\{\\} / { members }",`

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+			"val": ". digitsr",
+			"key": "frac"
+		}, {
+			"val": "digit / digit1-9 digits  / - digit / - digit1-9 digits",
+			"key": "int"
+		}, {
+			"val": "int / int frac / int exp / int frac exp",
+			"key": "number"
+		}, {
+			"val": "any-Unicode-character-except-\"-or-\\-or-control-character -or- \\\" -or- \\\\ -or- \\/ -or- \\b -or- \\f -or- \\n -or- \\r -or- \\t -or- \\u -or- four-hex-digits",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: this too:
`"val": "any-Unicode-character-except-\\\\"-or-\\\\\\\\-or-control-character -or- \\\\\\\\\\\\" -or- \\\\\\\\\\\\\\\\ -or- \\\\\\\\\/ -or- \\\\\\\\b -or- \\\\\\\\f -or- \\\\\\\\n -or- \\\\\\\\r -or- \\\\\\\\t -or- \\\\\\\\u -or- four-hex-digits",`

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        "JSON"
+	],
+	"template_type": "code",
+	"section_order": [
+		"Data Types",
+		"JSON Elements"
+	],
+	"sections": {
+		"JSON Elements": [{
+			"val": "string / number / object / array / true / false / null / undefined",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: change to
`"val": "string \/ number \/ object \/ array \/ true \/ false \/ null \/ undefined",`

also update all other fields

**Comment by gautamkrishnar** [comment]
@vCillusion Welcome to DDG. Really happy to have you in our team. :smile:

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+		"sourceUrl": "http://www.json.org/"
+	},
+	"aliases": [
+		"JavaScript Object Notation",
+		"Basics of JSON",
+		"Basics of JavaScript Object Notation",
+		"JavaScript Object Notation Syntax",
+		"Basic Syntax of JSON",
+		"Basic Syntax of JavaScript Object Notation",
+        "JSON"
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: fix the indentation

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+			"val": "value / value , elements",
+			"key": "elements"
+		}, {
+			"val": "[] / [ elements ]",
+			"key": "array"
+		}, {
+			"val": "string : value",
+			"key": "pair"
+		}, {
+			"val": "pair / pair , members",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: `"val": "pair \/ pair , members",`

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+	"section_order": [
+		"Data Types",
+		"JSON Elements"
+	],
+	"sections": {
+		"JSON Elements": [{
+			"val": "string / number / object / array / true / false / null / undefined",
+			"key": "value"
+		}, {
+			"val": "value / value , elements",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: `"val": "value \/ value , elements",`

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+			"key": "pair"
+		}, {
+			"val": "pair / pair , members",
+			"key": "members"
+		}, {
+			"val": "{} / { members }",
+			"key": "object"
+		}],
+		"Data Types": [{
+			"val": "digit / digit digits",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: `"val": "digit \/ digit digits",`

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+			"key": "digits"
+		}, {
+			"val": "e / e+ / e- / E / E+ / E-",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: "val": "e \/ e+ \/ e- \/ E \/ E+ \/ E-",

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+			"key": "e"
+		}, {
+			"val": "e digits",
+			"key": "exp"
+		}, {
+			"val": ". digitsr",
+			"key": "frac"
+		}, {
+			"val": "digit / digit1-9 digits  / - digit / - digit1-9 digits",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: `"val": "digit \/ digit1-9 digits  \/ - digit \/ - digit1-9 digits",`

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+			"key": "int"
+		}, {
+			"val": "int / int frac / int exp / int frac exp",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: `"val": "int \/ int frac \/ int exp \/ int frac exp",`

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+			"key": "number"
+		}, {
+			"val": "any-Unicode-character-except-\"-or-\\-or-control-character -or- \\\" -or- \\\\ -or- \\/ -or- \\b -or- \\f -or- \\n -or- \\r -or- \\t -or- \\u -or- four-hex-digits",
+			"key": "char"
+		}, {
+			"val": "char / char chars",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: `"val": "char \/ char chars",`

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+			"key": "chars"
+		}, {
+			"val": "\"\" / \" chars \"",
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/json-syntax.json]: `"val": "\"\" / \" chars \"",`

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 3 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: This one seems already escaped
**original **
`any-Unicode-character-except-"-or-\-or-control-character-or-\"-or-\\-or-\/-or-\b-or-\f-or-\n-or-\r-or-\t-or-\u-or-four-hex-digits
`**escaped**
`any-Unicode-character-except-\"-or-\\-or-control-character-or-\\\"-or-\\\\-or-\\\/-or-\\b-or-\\f-or-\\n-or-\\r-or-\\t-or-\\u-or-four-hex-digits`

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 5 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: fixed

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 4 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: done

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 1 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: done

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 7 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: done

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 6 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: done

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 2 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: done

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 8 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: done

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 9 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: done

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 10 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: done

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 11 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: done

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 3 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: done for \/

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 12 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: done

File: share/goodie/cheat_sheets/json/json-syntax.json (hunk 13 above)
Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: Already indented done for \/

File: share/goodie/cheat_sheets/json/json-syntax.json [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+    "sourceUrl": "http://www.json.org/"
+  },
+  "aliases": [
+    "JavaScript Object Notation",
+    "Basics of JSON",
+    "Basics of JavaScript Object Notation",
+    "JavaScript Object Notation Syntax",
+    "Basic Syntax of JSON",
+    "Basic Syntax of JavaScript Object Notation",
+    "JSON"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/json-syntax.json]: All the aliases should be lowercase.

Also, remove any of the aliases with `basic` or `syntax` in - the aliases should be strictly additional names.

Comment (vCillusion) [inline share/goodie/cheat_sheets/json/json-syntax.json]: @GuiltyDolphin makes sense.. but is there any other way to trigger for json basics or json syntax as of now other than this..

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=JSON syntax Cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/json_syntax_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@vCillusion Thanks! This LGTM :+1:

**Pull request merged**